src/rdkafka.c

```diff
@@ -188,22 +189,29 @@ void rd_kafka_log_buf (const rd_kafka_t *rk, int level, const char *fac,
                 rko->rko_u.log.str = rd_strdup(buf);
                 rd_kafka_q_enq(rk->rk_logq, rko);
 
-        } else if (rk->rk_conf.log_cb) {
-                rk->rk_conf.log_cb(rk, level, fac, buf);
+        } else if (conf->log_cb) {
+                conf->log_cb(rk, level, fac, buf);
         }
 }
 
-void rd_kafka_log0 (const rd_kafka_t *rk, const char *extra, int level,
-		   const char *fac, const char *fmt, ...) {
+/**
+ * @brief Logger
+ *
+ * @remark conf must be set, but rk may be NULL
```

Comment:
I generally try to avoid checking/asserting for obvious and trivial stuff (they seldom pay off),b but reserve the right to assert to the more stateful checks..